src/CLI/Progress.php

```diff
@@ -0,0 +1,17 @@
+<?php
+declare(strict_types=1);
+
+namespace Arkitect\CLI;
+
+use Arkitect\ClassSet;
+
+interface Progress
```

Comment:
every steps method in this interface represents a different steps in analysis. Concrete implementations can decide which information to show and how